### PR TITLE
feat: [SMP-2292]: Add SSL environment variables and volumes for redis

### PR DESCRIPTION
**Pull request by kapilgarg1996**
## Describe your changes
- Currently we support SSL environment variables only for events framework redis database and not for cache/lock databases.
- In SMP, customers tend to use a single redis instance for all use cases hence if SSL is enabled in DB, it will be enabled for events, cache and lock.
- To achieve this, I have added environment variables to enable/disable SSL for other use cases. 
- The changes are mainly in a service's `replace_configs.sh`, `deployment.yaml` and `values.yaml`
- The flow is like this : customer will update `values.yaml` to enable/disable ssl. Based on values set, environment variables will be added in `deployment.yaml` through helm templates. `replace_configs.sh` will use those env variables to modify service configs.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56862)

### CODE REVIEW

**Review by arya-harness — APPROVED** [review]

**Review by yogesh-chauhan — DISMISSED** [review]

**Review by archit-harness — DISMISSED** [review]

File: platform-service/chart/templates/deployment.yaml [hunk 1]
```diff
@@ -82,16 +82,22 @@ spec:
...
           {{- include "harnesscommon.dbv3.mongoConnectionEnv" (dict "ctx" $ "database" "resource-groups" "connectionURIVariableName" "RESOURCE_GROUP_MONGO_URI") | indent 10 }}
+          {{- include "harnesscommon.dbv3.sslEnv" (dict "context" $ "dbType" "redis" "dbName" "events" "variableNames" ( dict "sslEnabled" "EVENTS_FRAMEWORK_REDIS_SSL_ENABLED" "sslCATrustStorePath" "EVENTS_FRAMEWORK_REDIS_SSL_CA_TRUST_STORE_PATH" "sslCATrustStorePassword" "EVENTS_FRAMEWORK_REDIS_SSL_CA_TRUST_STORE_PASSWORD")) | indent 10 }}
+          {{- include "harnesscommon.dbv3.sslEnv" (dict "context" $ "dbType" "redis" "dbName" "lock" "variableNames" ( dict "sslEnabled" "LOCK_CONFIG_REDIS_SSL_ENABLED" "sslCATrustStorePath" "LOCK_CONFIG_REDIS_SSL_CA_TRUST_STORE_PATH" "sslCATrustStorePassword" "LOCK_CONFIG_REDIS_SSL_CA_TRUST_STORE_PASSWORD" )) | indent 10 }}
+          {{- include "harnesscommon.dbv3.sslEnv" (dict "context" $ "dbType" "redis" "dbName" "cache" "variableNames" ( dict "sslCATrustStorePath" "LOCK_CONFIG_REDIS_SSL_CA_TRUST_STORE_PATH" "sslCATrustStorePassword" "LOCK_CONFIG_REDIS_SSL_CA_TRUST_STORE_PASSWORD" )) | indent 10 }}
```
Comment (Jimit09Gandhi) [inline platform-service/chart/templates/deployment.yaml:87]: @kapilgarg1996 Where did u find usage of cache in PL service?

Comment (kapilgarg1996) [reply platform-service/chart/templates/deployment.yaml:87]: yepp, there isn't any cache db, hence the environment variable names are same for lock and cache but its redundant. I will remove this

**Review by yogesh-chauhan — DISMISSED** [review]

**Review by Jimit09Gandhi — DISMISSED** [review]

**Review by puthrayaharness — DISMISSED** [review]

**Review by harshjain12 — APPROVED** [review]

**Review by reetika04 — DISMISSED** [review]

**Comment by kapilgarg1996** [comment]
trigger uts

**Comment by kapilgarg1996** [comment]
trigger ut8

**Comment by kapilgarg1996** [comment]
trigger uts

**Comment by kapilgarg1996** [comment]
trigger uts

**Comment by kapilgarg1996** [comment]
trigger uts

**Comment by akshay-khole** [comment]
Force merging , approval were overwritten because of force push of commit

**Comment by kapilgarg1996** [comment]
Unrelated tests are failing. Reference: https://harness.slack.com/archives/C01JMN5P7EX/p1705034733396339

**Pull request merged**